### DISCUSSION TITLE
根据官方文档提供的定制主题的changeTheme方法，放入项目但是并不能实现主题的更改

### DISCUSSION THREAD

**Discussion by TJxiaobao** [ng-alain/ng-alain#2491] [Q&A]
本人是一名后端开发者，第一次接触前端，如果有不足还请见谅。
问题：在使用ng-alain 时，按照官方文档实现 主题切换：https://ng-alain.com/theme/customize-theme/zh
也是把 changeTheme 方法放入到了代码中（如下图）。但是在项目运行时并没有出现主题的更变。
希望能得到大佬们的解惑，那将不胜感激！
![image](https://github.com/ng-alain/ng-alain/assets/85919258/dc715313-cd70-403b-9495-3a81a7b34df6)
![image](https://github.com/ng-alain/ng-alain/assets/85919258/4552f020-fd75-4953-a9be-154385c1740a)
![image](https://github.com/ng-alain/ng-alain/assets/85919258/7d07d7c3-aef9-428b-b24b-c7e13211bde9)

**Comment by cipchk** [comment]
你这里的 `style` 创建后并没有写入到 `head` 里面。

```diff
style.href = 'assets/style.dark.css';
+ document.head.appendChild(style);
```

这算是文档上的错误 ，不过这里还涉及到 移除问题，我会修正这部分文档 。

可以参考：https://github.com/ng-alain/delon/pull/1784/files 写法。

**Reply by TJxiaobao** [reply]
是的大佬，您说的比较对，当我添加这一行时确实能渲染了，但是又出现了一个问题想向您请教。
因为这个主题更改我们打算通过和后端来一切协调更改，通过前端向后端发起保存主题字段，然后每一次保存都会重新刷新这个页面，这个时候就出现了问题，主题的渲染突然就变成了一瞬间，好像并没有一个持久化。面对这个问题想请教您这个有什么办法解决吗？
![image](https://github.com/ng-alain/ng-alain/assets/85919258/e65930bb-679d-4fe6-8b6a-7182c66e261f)
![image](https://github.com/ng-alain/ng-alain/assets/85919258/fdbe4f64-d430-4b1c-98bc-d32f60c17dc6)

**Reply by cipchk** [reply]
的 确如此，这是目前无法避免的，因为 `.css` 需要加载时间。

**Reply by TJxiaobao** [reply]
好的，谢谢您的指点。祝您生活愉快。
